### PR TITLE
tldr: update links for source

### PR DESCRIPTION
**Pull request by owenvoke**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by sbrl** [comment]
This PR updates the tldr C client links to the new repository, as said repository was recently renamed.

**Comment by owenvoke** [comment]
This appears to be failing due to the change to the SHA256 hash. I updated this as the package isn't building from source even with the currently live version.
```bash
brew install --build-from-source tldr
```

**Comment by sbrl** [comment]
For reference, both myself and @owenvoke are admins of the tldr-pages organisation on GitHub.

**Comment by SMillerDev** [comment]
If it isn't currently building, was the version retagged? And if so, why?

**Comment by owenvoke** [comment]
As far as I know, no. 🤔 The formula hash hasn't changed for years (https://github.com/Homebrew/homebrew-core/commit/3b3786d006), however it looks like the release in GitHub (https://github.com/tldr-pages/tldr-c-client/releases/tag/v1.3.0) was after that commit. So maybe? 🤔

Unfortunately I wasn't maintaining the project back then.

File: Formula/tldr.rb [hunk 1]
```diff
@@ -1,10 +1,10 @@
 class Tldr < Formula
   desc "Simplified and community-driven man pages"
   homepage "https://tldr.sh/"
-  url "https://github.com/tldr-pages/tldr-cpp-client/archive/v1.3.0.tar.gz"
-  sha256 "6210ece3f5d8f8e55b404e2f6c84be50bfdde9f0d194a271bce751a3ed6141be"
+  url "https://github.com/tldr-pages/tldr-c-client/archive/v1.3.0.tar.gz"
+  sha256 "7e7f67f4c3cf7d448847e837df2122069b0cc8f7ed6963431e914b7929655efe"
   revision 2
```
Comment (bayandin) [inline Formula/tldr.rb:6]:
```suggestion
```

Comment (bayandin) [reply Formula/tldr.rb:6]: Ah, it's just a link update.. nevermind then

**Review by dawidd6 — APPROVED** [review]

**Comment by owenvoke** [comment]
Thanks @dawidd6 👍

**Pull request closed without merging**